server/src/test/java/org/candlepin/async/tasks/RefreshPoolsForProductJobTest.java

```diff
@@ -0,0 +1,108 @@
+import org.candlepin.model.ProductCurator;
+import org.candlepin.pinsetter.tasks.BaseJobTest;
+import org.candlepin.service.OwnerServiceAdapter;
+import org.candlepin.service.SubscriptionServiceAdapter;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class RefreshPoolsForProductJobTest extends BaseJobTest {
+
+    private static final String VALID_ID = "valid_id";
+    private static final String VALID_NAME = "valid_name";
+    private static final String INVALID_ID = "";
+    private ProductCurator productCurator;
+    private PoolManager poolManager;
+    private SubscriptionServiceAdapter subAdapter;
+    private OwnerServiceAdapter ownerAdapter;
+
+    @Before
+    public void setUp() {
+        productCurator = mock(ProductCurator.class);
+        poolManager = mock(PoolManager.class);
+        subAdapter = mock(SubscriptionServiceAdapter.class);
+        ownerAdapter = mock(OwnerServiceAdapter.class);
+    }
+
+    @Test
+    public void shouldSucceed() throws Exception {
+        final String expected = "Pools refreshed for product: " + VALID_ID + "\n";
+        final AsyncJob job = new RefreshPoolsForProductJob(
+            productCurator, poolManager, subAdapter, ownerAdapter);
+        final Product product = new Product(INVALID_ID, VALID_NAME);
+        product.setUuid(VALID_ID);
+        final JobConfig jobConfig = RefreshPoolsForProductJob.createJobConfig()
+            .setTarget(product)
+            .setLazy(false);
+        final JobExecutionContext context = mock(JobExecutionContext.class);
+        doReturn(jobConfig.getJobArguments()).when(context).getJobArguments();
+        doReturn(product).when(productCurator).get(eq(VALID_ID));
+        doReturn(mock(Refresher.class)).when(poolManager).getRefresher(any(), any(), anyBoolean());
+
+        final Object actualResult = job.execute(context);
+
+        assertEquals(expected, actualResult);
+    }
+
+    @Test
+    public void shouldFailWhenNoProduct() throws Exception {
+        final AsyncJob job = new RefreshPoolsForProductJob(
+            productCurator, poolManager, subAdapter, ownerAdapter);
+        final Product product = new Product(INVALID_ID, VALID_NAME);
+        final JobConfig jobConfig = RefreshPoolsForProductJob.createJobConfig()
+            .setTarget(product)
+            .setLazy(false);
+        final JobExecutionContext context = mock(JobExecutionContext.class);
+        doReturn(jobConfig.getJobArguments()).when(context).getJobArguments();
+
+        final Object actualResult = job.execute(context);
+
+        assertEquals(
+            "Unable to refresh pools for product \"null\": Could not find a product with the specified UUID",
+            actualResult);
+    }
+
+    @Test(expected = JobConfigValidationException.class)
+    public void productUuidMustBePresent() throws JobConfigValidationException {
```

Comment:
One more small test for the happy path of `.validate` would be good (e.g. that product with a proper uuid and lazy flag are both set and no exception thrown.

Reply:
Done.